pkg/serverless/trace/trace.go

```diff
@@ -99,7 +102,13 @@ func (s *ServerlessTraceAgent) Start(enabled bool, loadConfig Load, lambdaSpanCh
 			context, cancel := context.WithCancel(context.Background())
 			tc.Hostname = ""
 			tc.SynchronousFlushing = true
-			s.ta = agent.NewAgent(context, tc, telemetry.NewNoopCollector(), &statsd.NoOpClient{})
+			var compressor compression.Component
+			if tc.HasFeature("zstd-encoding") {
+				compressor = zstd.NewComponent()
+			} else {
+				compressor = gzip.NewComponent()
+			}
```

Comment:
is this something we want to allow customers to tweak in the serverless environment. Would it be easier to use the same option always?

Reply:
This is actually an undocumented setting, that is meant to be hidden from the customer. We want to silently provide this option for us to gradually enable zstd on org2 prod clusters, and ensure no issue arises. If all goes well, the goal is to flip the setting in the next release, and use zstd by default in all cases.